api/c/indigo/src/indigo_molecule.cpp

```diff
@@ -58,7 +59,22 @@ bool IndigoBaseMolecule::is(const IndigoObject& object)
 {
     int type = object.type;
 
-    if (type == MOLECULE || type == QUERY_MOLECULE || type == REACTION_MOLECULE || type == SCAFFOLD || type == RGROUP_FRAGMENT || type == RDF_MOLECULE || type == SMILES_MOLECULE || type == CML_MOLECULE || type == JSON_MOLECULE || type == CDX_MOLECULE || type == SUBMOLECULE)
+    const static std::unordered_set<int> myTypes = 
+    { 
+        MOLECULE, 
+        QUERY_MOLECULE, 
+        REACTION_MOLECULE, 
+        SCAFFOLD, 
+        RGROUP_FRAGMENT, 
+        RDF_MOLECULE, 
+        SMILES_MOLECULE, 
+        CML_MOLECULE, 
+        JSON_MOLECULE, 
+        CDX_MOLECULE, 
+        SUBMOLECULE
+    };
+
+    if (myTypes.count(type))
```

Comment:
I agree that old "if" was huge and it could be improved, but not with a price of creating a new structure. Maybe we should do it with `switch`:
```c++
switch (object.type)
{
case MOLECULE:
case QUERY_MOLECULE:
case REACTION_MOLECULE:
...
    return true;
case ARRAY_ELEMENT:
    return is(((IndigoArrayElement&)object).get());
default:
    return false;
}
```

Reply:
Why not? This is normal trick to decrease complexity. (BTW, switch always significantly increases complexity)
`if` here answers the question "Does type belongs to one of the predefined values?" or "Does it belongs to the set of predefined values?"
Since it is const and static, initialization does not consume time and few memory. The search speed is quick.

Reply:
OK, how about plain array and plain cycle (sacrificing run-time search speed)?
```
	const static int myTypes[] =
	{
		MOLECULE,
		QUERY_MOLECULE,
		REACTION_MOLECULE,
		SCAFFOLD,
		RGROUP_FRAGMENT,
		RDF_MOLECULE,
		SMILES_MOLECULE,
		CML_MOLECULE,
		JSON_MOLECULE,
		CDX_MOLECULE,
		SUBMOLECULE
	};

	for (const int t : myTypes)
		if (type == t)
			return true;
```

Reply:
To be honest, I don't get the "switch always significantly increases complexity". It implements exactly the same logic with less amount of code (355 symbols):
```c++
switch (object.type)
{
case MOLECULE:
case QUERY_MOLECULE:
case REACTION_MOLECULE:
case SCAFFOLD:
case RGROUP_FRAGMENT:
case RDF_MOLECULE:
case SMILES_MOLECULE:
case CML_MOLECULE:
case JSON_MOLECULE:
case CDX_MOLECULE:
case SUBMOLECULE:
    return true;
case ARRAY_ELEMENT:
    return is(((IndigoArrayElement&)object).get());
default:
    return false;
}
``` than original (455 symbols) or your (415 symbols). What is more complex in it from your perspective?

Reply:
changed to switch operator